lib/pbench/agent/tool_data_sink.py

```diff
@@ -1609,71 +1831,225 @@ def put_document(self, data_ctx, hostname):
+            cp_path,
+            param_key,
+            params,
+            optional_md,
+            logger=logger,
+        )
+
+
+def main(argv):
+    _prog = Path(argv[0])
+    PROG = _prog.name
+    # The Tool Data Sink executable is in:
+    #   ${pbench_bin}/util-scripts/tool-meister/pbench-tool-data-sink
+    # So .parent at each level is:
+    #   _prog       ${pbench_bin}/util-scripts/tool-meister/pbench-tool-data-sink
+    #     .parent   ${pbench_bin}/util-scripts/tool-meister
+    #     .parent   ${pbench_bin}/util-scripts
+    #     .parent   ${pbench_bin}
+    pbench_bin = _prog.parent.parent.parent
+
     try:
         redis_host = argv[1]
         redis_port = argv[2]
         param_key = argv[3]
     except IndexError as e:
-        logger.error("Invalid arguments: %s", e)
+        print(f"{PROG}: Invalid arguments: {e}", file=sys.stderr)
         return 1
+    else:
+        if not redis_host or not redis_port or not param_key:
+            print(f"{PROG}: Invalid arguments: {argv!r}", file=sys.stderr)
+            return 1
+    try:
+        daemonize = argv[4]
+    except IndexError:
+        daemonize = "no"
+    else:
+        if not daemonize:
+            daemonize = "no"
 
-    global tar_path
     tar_path = find_executable("tar")
     if tar_path is None:
-        logger.error("External 'tar' executable not found")
+        print("External 'tar' executable not found", file=sys.stderr)
         return 2
 
-    global cp_path
     cp_path = find_executable("cp")
     if cp_path is None:
-        logger.error("External 'cp' executable not found")
+        print("External 'cp' executable not found", file=sys.stderr)
         return 2
 
+    try:
+        pbench_run = os.environ["pbench_run"]
+    except KeyError:
+        print(
+            "Unable to fetch pbench_run environment variable", file=sys.stderr,
+        )
+        return 3
+
     try:
         redis_server = redis.Redis(host=redis_host, port=redis_port, db=0)
     except Exception as e:
-        logger.error(
-            "Unable to connect to redis server, %s:%s: %s", redis_host, redis_port, e
+        print(
+            f"Unable to connect to redis server, {redis_host}:{redis_port}: {e}",
+            file=sys.stderr,
         )
         return 4
 
     try:
         hostname = os.environ["_pbench_full_hostname"]
     except KeyError:
-        logger.error("Unable to fetch _pbench_full_hostname environment variable")
-        return 4
+        print(
+            "Unable to fetch _pbench_full_hostname environment variable",
+            file=sys.stderr,
+        )
+        return 5
 
     try:
-        params_raw = redis_server.get(param_key)
-        if params_raw is None:
-            logger.error('Parameter key, "%s" does not exist.', param_key)
-            return 5
-        logger.debug("params_key (%s): %r", param_key, params_raw)
-        params_str = params_raw.decode("utf-8")
+        # Loop waiting for the key to show up.
```

Comment:
"Wait for the parameter message to show up."  (We have the key, and there's no loop here -- _how_ we wait is an implementation detail of the function that we're about to call....)